jaxley/utils/cell_utils.py

```diff
@@ -291,9 +294,11 @@ def interpolate_xyz(loc: float, coords: np.ndarray):
     Return:
         Interpolated xyz coordinate at `loc`, shape `(3,).
     """
-    return vmap(lambda x: jnp.interp(loc, jnp.linspace(0, 1, len(x)), x), in_axes=(1,))(
-        coords[:, :3]
-    )
+    dl = np.sqrt(np.sum(np.diff(coords[:, :3], axis=0) ** 2, axis=1))
+    pathlens = np.insert(np.cumsum(dl), 0, 0) # cummulative length of sections
+    norm_pathlens = pathlens/pathlens[-1]  # path lengths normalized to [0,1]
```

Comment:
We instead want to interpolate along the traced out path, according to the pathlengths between the coords. Hence `xp` should correspond to the locs at each xyz coord.